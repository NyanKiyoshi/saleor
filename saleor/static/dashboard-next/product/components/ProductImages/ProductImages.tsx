import InfoIcon from "material-ui-icons/Info";
import Card, { CardContent } from "material-ui/Card";
import GridList, { GridListTile, GridListTileBar } from "material-ui/GridList";
import IconButton from "material-ui/IconButton";
import Subheader from "material-ui/List/ListSubheader";
import { withStyles } from "material-ui/styles";
import Typography from "material-ui/Typography";
import * as React from "react";

import { CircularProgress } from "material-ui/Progress";
import PageHeader from "../../../components/PageHeader";
import i18n from "../../../i18n";

interface ProductImagesProps {
  imageList?: Array<{
    id: string;
    alt: string;
    url: string;
    order: number;
  }>;
}

const decorate = withStyles(theme => ({
  root: {
    display: "grid",
    gridTemplateColumns: "repeat(4, 1fr)",
    gridColumnGap: `${theme.spacing.unit * 2}px`,
    gridRowGap: `${theme.spacing.unit * 2}px`,
    [theme.breakpoints.down("md")]: {
      gridTemplateColumns: "repeat(3, 1fr)"
    },
    [theme.breakpoints.down("sm")]: {
      gridTemplateColumns: "repeat(2, 1fr)"
    }
  },
  card: {
    marginTop: theme.spacing.unit * 2,
    [theme.breakpoints.down("sm")]: {
      marginTop: 0
    }
  },
  gridElement: {
    "& img": {
      width: "100%"
    }
  }
}));

export const ProductImages = decorate<ProductImagesProps>(
  ({ classes, imageList }) => (
    <Card className={classes.card}>
      <PageHeader title={i18n.t("Images")} />
      <CardContent>
        <div className={classes.root}>
          {imageList === undefined || imageList === null ? (
            <CircularProgress />
          ) : imageList.length > 0 ? (
            imageList
              .sort((prev, next) => (prev.order > next.order ? 1 : -1))
              .map(tile => (
                <GridListTile
                  key={tile.id}
                  className={classes.gridElement}
                  component="div"
                >
                  <img src={tile.url} alt={tile.alt} />
                  <GridListTileBar
                    title={tile.alt || i18n.t("No description")}
                  />
                </GridListTile>
              ))
          ) : (
            <Typography>{i18n.t("No images available")}</Typography>
          )}
        </div>
      </CardContent>
    </Card>
  )
);
export default ProductImages;
