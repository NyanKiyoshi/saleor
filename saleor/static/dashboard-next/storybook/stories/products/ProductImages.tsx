import { storiesOf } from "@storybook/react";
import * as React from "react";

import * as placeholder from "../../../../images/placeholder255x255.png";
import ProductImages from "../../../products/components/ProductImages";
import { images as imagesFixture } from "../../../products/fixtures";

const images = imagesFixture.map(image => ({ ...image, url: placeholder }));

storiesOf("Products / ProductImages", module)
  .add("without data", () => <ProductImages imageList={[]} />)
  .add("with data", () => <ProductImages imageList={images} />)
  .add("when loading data", () => <ProductImages placeholder={placeholder} />);
