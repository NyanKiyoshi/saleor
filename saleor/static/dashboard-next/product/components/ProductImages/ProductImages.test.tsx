import * as React from "react";
import * as renderer from "react-test-renderer";

import ProductImages from "./";

const placeholder = "url/to/image.png";
const images = [
  {
    id: "2",
    alt: "Image 2",
    url: placeholder,
    order: 2
  },
  {
    id: "1",
    alt: "Image 1",
    url: placeholder,
    order: 1
  }
];

describe("<ProductImages />", () => {
  it("renders without data", () => {
    const component = renderer.create(<ProductImages imageList={[]} />);
    expect(component).toMatchSnapshot();
  });
  it("renders with data", () => {
    const component = renderer.create(<ProductImages imageList={images} />);
    expect(component).toMatchSnapshot();
  });
  it("renders when loading data", () => {
    const component = renderer.create(<ProductImages />);
    expect(component).toMatchSnapshot();
  });
});
