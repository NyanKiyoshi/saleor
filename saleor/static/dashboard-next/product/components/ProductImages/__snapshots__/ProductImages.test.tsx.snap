// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImages /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 Component-card-2"
>
  <div
    className="Component-root-34"
  >
    <h2
      className="MuiTypography-root-37 MuiTypography-title-43 Component-title-36"
    >
      Images
    </h2>
    <div
      className="Component-action-31"
    />
  </div>
  <div
    className="MuiCardContent-root-61"
  >
    <div
      className="Component-root-1"
    >
      <div
        className="MuiCircularProgress-root-77 MuiCircularProgress-colorPrimary-78"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg-80 MuiCircularProgress-svgIndeterminate-81"
          viewBox="0 0 50 50"
        >
          <circle
            className="MuiCircularProgress-circle-82 MuiCircularProgress-circleIndeterminate-83"
            cx={25}
            cy={25}
            fill="none"
            r={20}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductImages /> renders with data 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 Component-card-2"
>
  <div
    className="Component-root-34"
  >
    <h2
      className="MuiTypography-root-37 MuiTypography-title-43 Component-title-36"
    >
      Images
    </h2>
    <div
      className="Component-action-31"
    />
  </div>
  <div
    className="MuiCardContent-root-61"
  >
    <div
      className="Component-root-1"
    >
      <div
        className="MuiGridListTile-root-62 Component-gridElement-3"
      >
        <div
          className="MuiGridListTile-tile-63"
        >
          <img
            alt="Image 1"
            src="url/to/image.png"
          />
          <div
            className="MuiGridListTileBar-root-66 MuiGridListTileBar-titlePositionBottom-67"
          >
            <div
              className="MuiGridListTileBar-titleWrap-70"
            >
              <div
                className="MuiGridListTileBar-title-73"
              >
                Image 1
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGridListTile-root-62 Component-gridElement-3"
      >
        <div
          className="MuiGridListTile-tile-63"
        >
          <img
            alt="Image 2"
            src="url/to/image.png"
          />
          <div
            className="MuiGridListTileBar-root-66 MuiGridListTileBar-titlePositionBottom-67"
          >
            <div
              className="MuiGridListTileBar-titleWrap-70"
            >
              <div
                className="MuiGridListTileBar-title-73"
              >
                Image 2
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductImages /> renders without data 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 Component-card-2"
>
  <div
    className="Component-root-34"
  >
    <h2
      className="MuiTypography-root-37 MuiTypography-title-43 Component-title-36"
    >
      Images
    </h2>
    <div
      className="Component-action-31"
    />
  </div>
  <div
    className="MuiCardContent-root-61"
  >
    <div
      className="Component-root-1"
    >
      <p
        className="MuiTypography-root-37 MuiTypography-body1-46"
      >
        No images available
      </p>
    </div>
  </div>
</div>
`;
